prow/gcpproxy-presubmit.sh

```diff
@@ -25,83 +25,31 @@ pwd)
 ROOT=$(dirname "$WD")
 export PATH=$PATH:$GOPATH/bin
 
+
 gcloud config set core/project cloudesf-testing
+gcloud auth activate-service-account \
+  --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
 
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 cd "${ROOT}"
 . ${ROOT}/scripts/all-utilities.sh || { echo 'Cannot load Bash utilities';
 exit 1; }
+. ${ROOT}/tests/e2e/scripts/prow-utilities.sh || { echo 'Cannot load Bash utilities';
+exit 1; }
 
-echo '======================================================='
-echo '===================== Setup Cache ====================='
-echo '======================================================='
-try_setup_bazel_remote_cache "${PROW_JOB_ID}" "${IMAGE}" "${ROOT}" "presubmit-${PRESUBMIT_TEST_CASE}"
-
-echo '======================================================='
-echo '===================== Spelling Check ====================='
-echo '======================================================='
-make spelling.check
-
-# golang test
-echo '======================================================'
-echo '=====================   Go test  ====================='
-echo '======================================================'
-if [ ! -d "$GOPATH/bin" ]; then
-  mkdir $GOPATH/bin
-fi
-if [ ! -d "bin" ]; then
-  mkdir bin
-fi
-export GO111MODULE=on
-make tools
-make depend.install
-
-# GOOGLE_APPLICATION_CREDENTIALS will be set in our test environment but this env
-# var if set will be used by start_proxy.py as service_json_path.
- (unset GOOGLE_APPLICATION_CREDENTIALS; make test)
+wait_apiproxy_image
 
-# c++ test
-echo '======================================================'
-echo '===================== Bazel test ====================='
-echo '======================================================'
-if [ -z ${PRESUBMIT_TEST_CASE} ];
-then
-  echo "running normal presubmit test"
-else
-  echo "running ${PRESUBMIT_TEST_CASE} presubmit test"
-fi
+download_envoy_binary
+chmod +x ${ROOT}/bin/envoy
+VERSION=$(cat ${ROOT}/VERSION)
 
-case "${PRESUBMIT_TEST_CASE}" in
-  "asan")
-    make test-envoy-asan
-    ;;
-  "msan")
-    make test-envoy-msan
-    ;;
-  "tsan")
-    make test-envoy-tsan
-    ;;
-  *)
-    make test-envoy
-    ;;
-esac
+SHA=$(git rev-list -n 1 $(git describe --abbrev=0))
```

Comment:
Are you directly change our presubmit test to be api-compatibilty test?   I hope not.  
It should be a separate test that we can ignore for some PR

Reply:
Of course not. I just use gcpproxy-presubmit as a test trigger before we config the prow.